upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -824,6 +825,37 @@ func (tf *TemplateFunctions) GetNodeInstanceGroups() map[string]kops.InstanceGro
 	return nodegroups
 }
 
+type ClusterAutoscalerNodeGroup struct {
+	AutoScale *bool
+	MinSize   int32
+	MaxSize   int32
+	Other     string
+}
+
+// GetClusterAutoscalerGroups returns a map containing ClusterAutoscaler info for each instance group of type Node.
+func (tf *TemplateFunctions) GetClusterAutoscalerNodeGroups() map[string]ClusterAutoscalerNodeGroup {
+	cluster := tf.Cluster
+	groups := make(map[string]ClusterAutoscalerNodeGroup)
+	for _, ig := range tf.KopsModelContext.InstanceGroups {
+		if ig.Spec.Role == kops.InstanceGroupRoleNode && (ig.Spec.Autoscale == nil || fi.ValueOf(ig.Spec.Autoscale)) {
+			group := ClusterAutoscalerNodeGroup{
+				AutoScale: ig.Spec.Autoscale,
+				MinSize:   fi.ValueOf(ig.Spec.MinSize),
+				MaxSize:   fi.ValueOf(ig.Spec.MaxSize),
+			}
+			if cluster.Spec.GetCloudProvider() == kops.CloudProviderGCE {
+				cloud := tf.cloud.(gce.GCECloud)
+				format := "https://www.googleapis.com/compute/v1/projects/%s/zones/%s/instanceGroups/%s"
```

Comment:
I've seen this output used in `kops get instances` as well, so wondering if there may be some generic function for this somewhere.

Reply:
Not to my knowledge...

Reply:
Looks like that same sprintf is used everywhere. But there is also a function here: upup/pkg/fi/cloudup/gcetasks/instance.go

But cleanup for another time, I suppose